nix/build-support/check-blueprints.nix

```diff
@@ -0,0 +1,29 @@
+{pkgs}: let
+  ld_library_path = import ./ld_library_path.nix {
+    inherit pkgs;
+    lib = pkgs.lib;
+    stdenv = pkgs.stdenv;
+  };
+  gi_typelib_path = import ./gi_typelib_path.nix {
+    inherit pkgs;
+    lib = pkgs.lib;
+    stdenv = pkgs.stdenv;
+  };
+  program = pkgs.writeShellApplication {
+    name = "check-blueprints";
+    runtimeInputs = [
+      pkgs.blueprint-compiler
+      pkgs.findutils
+    ];
+    text = ''
+      export LD_LIBRARY_PATH="${ld_library_path}";
+      export GI_TYPELIB_PATH="${gi_typelib_path}"
+
+      find src -name \*.blp -exec blueprint-compiler format {} \+
+      find src -name \*.blp -exec blueprint-compiler compile --output=/dev/null {} \;
+    '';
+  };
+in {
+  type = "app";
```

Comment:
Why does this have to be a whole Nix thing? Can't we just run the below?

```
find src -name \*.blp -exec blueprint-compiler format {} \+
find src -name \*.blp -exec blueprint-compiler compile --output=/dev/null {} \;
```

Should this be a simple shell script in `build-support` that others could run without Nix? (Although, the Nix devShell will give you the proper env vars). Our official build environment is Nix and I like using Nix for what it's good for, but I try to be pragmatic in balancing that between forcing it.

Reply:
I agree that we should really just have a shell script

Reply:
Other than I like Nix? No. Fixed.